nodes/src/nodes/tool_filesystem/IInstance.py

```
@@ -0,0 +1,353 @@
+            'additionalProperties': False,
+        },
+        description=('Read a file from the account file store and return its contents as a decoded string. Required: "path" (relative path). Optional: "encoding" (default "utf-8"). Returns: {path, content, size} where size is the byte length before decoding.'),
+    )
+    def read_file(self, args):
+        path, encoding, _ = self._prepare(args, 'read_file', needs_encoding=True)
+
+        data = _run_async(self.IGlobal.file_store.read(path))
+        try:
+            content = data.decode(encoding)
+        except UnicodeDecodeError as e:
+            raise ValueError(f'Failed to decode file {path!r} using encoding {encoding!r}: {e}') from e
```

Ruff (0.15.10)

[warning] Avoid specifying long messages outside the exception class

(TRY003)